internal/code/package.go

```diff
@@ -1,14 +1,68 @@
 package code
 
 import (
+	"errors"
 	"go/ast"
+	"go/parser"
+	"go/token"
 	"strings"
+
+	"golang.org/x/tools/go/packages"
+)
+
+var (
+	errNoPackagesFound = errors.New("no packages found")
 )
 
-// ParsePackage extracts package name, struct and interface implementations from
-// map[string]*ast.Package. Test files will be ignored.
-func ParsePackage(pkgs map[string]*ast.Package) (Package, error) {
+type dirparseFn func(dir string) (pkgs map[string]*ast.Package, err error)
```

Comment:
Return variable name is not needed here. (Remove from the default implementation also)

```suggestion
type parseDirFn func(dir string) (map[string]*ast.Package, error)
```